src/library/Waveform.js

```diff
@@ -44,45 +53,102 @@ class Waveform {
     /** @private @const {!dataArray} The array that the time domain
      * data will be copied to */
     this.dataArray_ = new Float32Array(this.analyser_.frequencyBinCount);
+        
+    // Initialize canvas with background grid
+    this.drawBackground_();
+  }
+
+  /**
+   * Draw the background
+   * @private
+   */
+  drawBackground_() {
+    // Fill with white background
+    this.canvasContext_.fillStyle = 'white';
+    this.canvasContext_.fillRect(0, 0, this.width_, this.height_);
   }
-  
+
   /**
    * Render the waveform on the canvas. This function is called
    * periodically to update the waveform display. This class doesn't
    * have a timer or a render loop; it requires an external driver for
    * the actual rendering.
    */
   draw() {
-    this.canvasContext_.fillStyle = 'red';
-    this.canvasContext_.fillRect(0, 0, 1, 1);
-
-    this.canvasContext_.clearRect(this.currentX_, 0, 1, this.height_);
-
+    // Get the time domain data
     this.analyser_.getFloatTimeDomainData(this.dataArray_);
+    
+    // Calculate the average amplitude with enhanced scaling for more dramatic Y-axis movement
     const average = this.dataArray_.reduce((a, b) => a + b) / this.dataArray_.length;
-
-    const currentY = (average + 1) * (this.height_ / 2);
-
-    this.canvasContext_.fillStyle = 'red';
-    this.canvasContext_.fillRect(this.currentX_ + 2, 0, 1, this.height_);
-
+    
+    // Apply amplitude multiplier for more dramatic vertical movement
+    const amplitudeMultiplier = 5.0; // Increase this value for more dramatic height
+    const scaledAmplitude = average * amplitudeMultiplier;
+    
+    // Calculate Y position with enhanced vertical range
+    const centerY = this.height_ / 2;
+    const currentY = centerY + (scaledAmplitude * (this.height_ / 2));
+    
+    // Clamp to canvas bounds to prevent drawing outside
+    const clampedY = Math.max(0, Math.min(this.height_, currentY));
+    
+    // Draw the waveform line
```

Comment:
Here we should draw vertical lines that represent the lowest and highest points the waveform reached in the time range represented by the data array.

We can calculate the minimum and maximum values, then scale them by the height of the waveform display.

Then we can stroke one vertical line at the current X position, between these maximum and minimum values.